### PR TITLE
docs(Objects in Ember): update intro to objects

### LINKED ISSUE #155: Write a friendlier introduction to the Objects guide
When someone is done with the tutorial, the first thing they read is [this](https://guides.emberjs.com/release/object-model/). It's very confusing to beginners and right away, makes us look dated, because ES2015 is hardly new 😆 

To-do:
- [ ] take out the reference to ES2015 being "new"
- [ ] write a better introduction. Imagine someone is reading this on day-one of Ember! New learners are probably the only people who look at this page.
- [ ] only make changes to the markdown file for the latest version of the guides

A more friendly introduction could hit these points, but these are just suggestions, so feel free to take a different direction:
- One of the first things you'll notice when you generate JavaScript files in Ember is that most of the code you will write goes inside of an object.
- While an Ember Object might look a lot like an ES2015 JavaScript class, it has some special powers.
- The most important is that an Ember Object can be watched for changes
- (description of what is meant here)

### PR DESCRIPTION
**Pull request by chrisrng**
Fixes #155 

[rendered](https://github.com/ember-learn/guides-source/blob/e655e9abc7b3720e6964560c79eb8a5fcb29c01d/guides/v3.4.0/object-model/index.md)

### CODE REVIEW

File: guides/v3.4.0/object-model/index.md [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
 
-Ember also extends the JavaScript `Array` prototype with its
-[Ember.Enumerable](https://emberjs.com/api/ember/2.15/classes/Ember.Enumerable) interface to provide change observation for arrays.
+Most objects in Ember, including Routes, Models, Services, Mixins, Controllers, and Components extend the `EmberObject` class.
 
-Finally, Ember extends the `String` prototype with a few [formatting and
-localization methods](https://www.emberjs.com/api/ember/release/classes/String).
+### Why Ember Objects?
+
+The most important reason is that an Ember Object can be watched for changes – or _observed_. Ember Objects, being [observable](https://www.emberjs.com/api/ember/release/classes/Observable), supports observing any of its properties including [computed properties](https://guides.emberjs.com/release/object-model/computed-properties/). It is one of the fundamental ways that models, controllers and views communicate with each other in an Ember application.
```
Comment (jenweber) [inline guides/v3.4.0/object-model/index.md]: Possible simplification of "Ember Objects, being observable...."

For example, being [observable](https://www.emberjs.com/api/ember/release/classes/Observable) is important for [computed properties](https://guides.emberjs.com/release/object-model/computed-properties/).

File: guides/v3.4.0/object-model/index.md [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
-JavaScript objects don't support the observation of property value changes.
-Consequently, if an object is going to participate in Ember's binding
-system you may see an `Ember.Object` instead of a plain object.
+### Introducing: Ember Objects
 
-[Ember.Object](https://www.emberjs.com/api/ember/release/modules/@ember%2Fobject) also provides a class system, supporting features like mixins
-and constructor methods. Some features in Ember's object model are not present in
-JavaScript classes or common patterns, but all are aligned as much as possible
-with the language and proposed additions.
+The [Ember Object](https://www.emberjs.com/api/ember/release/classes/EmberObject) class extends plain JavaScript objects to provide core framework functions such as participating in Ember's binding system.
```
Comment (jenweber) [inline guides/v3.4.0/object-model/index.md]: Can we add a very short definition of "binding system" here? Something like "... binding system, where changes to an object automatically trigger updates to the user interface."

File: guides/v3.4.0/object-model/index.md (hunk 1 above)
Comment (chrisrng) [reply guides/v3.4.0/object-model/index.md]: Fixed!

File: guides/v3.4.0/object-model/index.md (hunk 2 above)
Comment (chrisrng) [reply guides/v3.4.0/object-model/index.md]: Added this as well

**Comment by jenweber** [comment]
Hey, I missed one thing in my review. Links between topics within this app should use relative links, I.e. `../../topic/subtopic` rather than linking to URLs like guides.emberjs.com. Can you correct that? Thanks!

**Pull request merged**